backend/typescript/rest/schedulingRoutes.ts

```diff
@@ -119,10 +119,23 @@ schedulingRouter.put("/:id", updateSchedulingDtoValidator, async (req, res) => {
   }
 });
 
-/* Delete scheduling by id */
-schedulingRouter.delete("/:id", async (req, res) => {
+/* Delete scheduling by id (e.g. /scheduling/63)
+  and takes optional query for recurring donation 
+  id to delete by recurring donations (e.g. /scheduling/63?recurringDonationId=1)
+*/
+schedulingRouter.delete("/:id?", async (req, res) => {
   const { id } = req.params;
-  if (id) {
+  const { recurringDonationId } = req.query;
+  if (recurringDonationId) {
+    try {
+      await schedulingService.deleteSchedulingByRecurringDonationId(
+        recurringDonationId as string,
+      );
+      res.status(204).send();
+    } catch (error: unknown) {
+      res.status(500).json({ error: getErrorMessage(error) });
```

Comment:
if we want to avoid the case shehryar mentioned, you could do something like the following since we ideally want the request to either have just the param or just the query. But again, not really necessary I think we should merge it.

```typescript
if (id && recurringDonationId) {
 await sendResponseByMimeType(res, 400, contentType, [
      {
        error: "Cannot delete by both id and recurringSchedulingId",
      },
    ]);
    return;

}
```